qui/tray/disk_space.py

```diff
@@ -33,6 +33,14 @@ def __populate_pools(self):
                 self.warning_message.append(
                     "\n{:.1%} space left in pool {}".format(
                         1-pool.usage/pool.size, pool.name))
+            if 'metadata_size' in pool.usage_details:
+                metadata_usage = pool.usage_details['metadata_usage'] / \
+                                 pool.usage_details['metadata_size']
```

Comment:
1. Check for `metadata_size` being 0.
2. You won't like it, but you might also need to add `usage_details` property to `Pool` class in qubesadmin module (client side of Admin API). But the problem is at Admin API there is one call for all the info. And it lands in pool.config dict. I don't see any easy and reliable (independent of storage driver) way to extract "usage" info from other info. Some heuristic would be looking for "usage" and "size" substrings, but that's ugly :/ Maybe force some prefix in admin api call output? Or rename `config` to `info` (keeping compatibility wrapper) and don't distinguish them at all?

Reply:
what do you mean by 'might'?

Reply:
There are alternatives, listed in this comment.

Reply:
tbh I'm not sure what do you mean by the words there. is it documented anywhere in more detail?

Reply:
Admin API documentation and basic concepts: https://www.qubes-os.org/doc/admin-api/
Server side (qubesd service) lives in core-admin repo and you added relevant part here: https://github.com/QubesOS/qubes-core-admin/pull/278/files#diff-0c889398bb7fbe645fcda185af7132de
Client side (qubesadmin python module) lives in core-admin-client, documentation here: https://dev.qubes-os.org/projects/core-admin-client/en/latest/qubesadmin.html#qubesadmin.storage.Pool

In manager you have objects from qubesadmin module, client side of Admin API. Changes in server side Admin API in most cases require also adding relevant part to its client side.